workshop.pl

```diff
@@ -705,6 +707,47 @@ sub arg_handler {
     exit 15;
 }
 
+# copy over any files from any requirements
+foreach my $req (@{$active_requirements{'array'}}) {
+    if ($req->{'json'}{'type'} eq 'files') {
+        # currently a single, local file is supported
+        if ($req->{'json'}{'files_info'}{'type'} eq 'local-copy') {
+            if (-e $req->{'json'}{'files_info'}{'src'}) {
+                if (defined $req->{'json'}{'files_info'}{'dst'}) {
+                    ($command, $command_output, $rc) = run_command("/bin/cp -LR " .
+                                                        $req->{'json'}{'files_info'}{'src'} .
+                                                        " " . $container_mount_point . "/" .
+                                                        $req->{'json'}{'files_info'}{'dst'});
+                    if ($rc != 0) {
+                        logger('info', "failed\n", 1);
+                        command_logger('error', $command, $rc, $command_output);
+                        logger('error', "Failed to copy \"$req->{'json'}{'files_info'}{'src'}\" to the temporary container!\n");
+                        exit 35;
```

Comment:
Do you have a good idea for managing the return codes?  I like the idea of having unique return codes for each exit but manually managing the values is a real pain and leads to mistakes like 36 being used multiple times here.

Reply:
I think we will have to define vars at the beginning of all programs for error code types, like "my $error_copy_failed = 35" and reference that var.   I really hate having to update actual numbers within the code.

Reply:
Yeah, I agree.  Might as well just leave it as is for now and I can clean it up later.